### PR TITLE
Added documentation for methods ROS2, ROS3, ROS2PR, ROS2S, ROS3PR, Sc…

### PR DESCRIPTION
**Pull request by cwittens**
…holz47,ROS34PRw, ROS3PRL, ROS3PRL2

## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

added documentation for the methods added in https://github.com/SciML/OrdinaryDiffEq.jl/pull/2113, https://github.com/SciML/OrdinaryDiffEq.jl/pull/2126 and https://github.com/SciML/OrdinaryDiffEq.jl/pull/2165

### CODE REVIEW

File: docs/src/solvers/ode_solve.md [hunk 1]
```diff
@@ -550,6 +550,25 @@ to be thread safe. It parallelizes the `nlsolve` calls inside the method.
     4th order interpolant.
   - `Rodas5P` - A 5th order A-stable stiffly stable Rosenbrock method with a stiff-aware
     4th order interpolant. Has improved stability in the adaptive time stepping embedding.
+  - `ROS2` - A 2nd order L-stable Rosenbrock-Wanner method with 2 internal stages.
+  - `ROS3` - A 3rd order L-stable Rosenbrock-Wanner method with 3 internal stages
+    with an embedded strongly A-stable 2nd order method.
+  - `ROS2PR` - A 2nd order stiffly accurate Rosenbrock-Wanner method with 3 internal stages with Rinf=0.
+    For problems with medium stiffness the convergence behaviour is very poor
+    and it is recommended to use ROS2S instead.
```
Comment (ranocha) [inline docs/src/solvers/ode_solve.md:558]:
```suggestion
    and it is recommended to use `ROS2S` instead.
```

**Review by ranocha — COMMENTED** [review]
Thanks a lot! I guess we should wait until https://github.com/SciML/OrdinaryDiffEq.jl/pull/2165 is merged before merging this

**Comment by cwittens** [comment]
Maybe a question about the documentation in general. Does it make sense to put a hyperlink to the paper where the methods are discussed? Right now you can only find that link when looking up the documentation inside OrdinaryDiffEq using help, which seems counter intuitive to me.

**Comment by ranocha** [comment]
I would be in favor of adding such links. Maybe we could just print the docstring of each algorithm here?

**Comment by ChrisRackauckas** [comment]
They should go into the OrdinaryDiffEq docs (which should then get a section here when it's ready enough in the API section). This is supposed to just be a summary of available solvers, not a complete description of each one.

**Pull request merged**